apps/server/src/modules/school/repo/school.repo.ts

```diff
@@ -32,6 +32,26 @@ export class SchoolMikroOrmRepo extends BaseDomainObjectRepo<School, SchoolEntit
 		return schools;
 	}
 
+	public async getExternalSchools(
+		ownSchoolId: EntityId,
+		query: SchoolQuery,
+		options?: IFindOptions<SchoolProps>
+	): Promise<{ schools: School[]; count: number }> {
+		const scope = new SchoolScope();
+		scope.allowEmptyQuery(true);
+		scope.byFederalState(query.federalStateId);
+		scope.byExternalId(query.externalId);
+		scope.bySystemId(query.systemId);
```

Comment:
what is the reason to include `allowEmptyQuery`, `byExternalId` and `bySystemId`?
as I understand the `school-query.params` we only can hand over `federalStateId` additionally to the pagination params

Reply:
True. Refactored query-parameter to federalStateId but kept allowEmptyQuery(true) - being unsure about the implications of removing it.